### ISSUE TITLE
Proposal: warn on /index imports

### ISSUE THREAD

**Issue by mike1808** [benmosher/eslint-plugin-import#985]
My IDE likes to add `/index` to the import path when I specify some folder path with `index.js` file in it. For example:

```
- module
    - index.js

- file.js
```
```js
// file.js
import module from './module'
```

and it becomes

```js
// file.js
import module from './module/index'
```

Can we have a rule which disallows explicit specification of the `index` file?

**Comment by ljharb** [comment]
This seems like it'd be handled by #912 - since that could be shortened to `./module`, it would warn.
